src/CommunityToolkit.Maui.Core/Views/Popup/MauiPopup.macios.cs

```diff
@@ -0,0 +1,175 @@
+using Microsoft.Maui.Platform;
+using UIKit;
+
+namespace CommunityToolkit.Core.Views;
+
+/// <summary>
+/// The navite implementation of Popup control.
+/// </summary>
+public class MauiPopup : UIViewController
+{
+	readonly IMauiContext mauiContext;
+
+	/// <summary>
+	/// Constructor of <see cref="MauiPopup"/>.
+	/// </summary>
+	/// <param name="mauiContext">An instace of <see cref="IMauiContext"/>.</param>
+	/// <exception cref="ArgumentNullException">If <paramref name="mauiContext"/> is null an exception will be thrown. </exception>
+	public MauiPopup(IMauiContext mauiContext)
+	{
+		this.mauiContext = mauiContext ?? throw new ArgumentNullException(nameof(mauiContext));
+	}
+
+	/// <summary>
+	/// An instance of the <see cref="PageHandler"/> that holds the <see cref="IPopup.Content"/>.
+	/// </summary>
+	public PageHandler? Control { get; private set; }
+
+	/// <summary>
+	/// An instace of the <see cref="IPopup"/>.
+	/// </summary>
+	public IPopup? VirtualView { get; private set; }
+
+	internal UIViewController? ViewController { get; private set; }
+
+	/// <summary>
+	/// Method to update the Popup's size.
+	/// </summary>
+	/// <param name="size"></param>
+	public void SetElementSize(Size size) =>
+		Control?.ContainerView?.SizeThatFits(size);
+
+	/// <inheritdoc/>
+	public override void ViewDidLayoutSubviews()
+	{
+		base.ViewDidLayoutSubviews();
+
+		_ = View ?? throw new InvalidOperationException($"{nameof(View)} cannot be null");
+		SetElementSize(new Size(View.Bounds.Width, View.Bounds.Height));
+	}
+
+	/// <summary>
+	/// Method to initialize the native implementation.
+	/// </summary>
+	/// <param name="element">An instance of <see cref="IPopup"/>.</param>
+	public void SetElement(IPopup element)
+	{
+		var mainPage = Application.Current?.MainPage ?? throw new InvalidOperationException($"{nameof(Application.Current.MainPage)} cannot be null");
+
+		VirtualView = element;
+		ModalPresentationStyle = UIModalPresentationStyle.Popover;
+
+		_ = View ?? throw new InvalidOperationException($"{nameof(View)} cannot be null");
+		_ = VirtualView ?? throw new InvalidOperationException($"{nameof(VirtualView)} cannot be null.");
+
+		Control = CreateControl(VirtualView);
+		ViewController ??= CreateViewController(mainPage);
+
+		SetPresentationController();
+		SetView(View, Control);
+		AddToCurrentPageViewController(ViewController, VirtualView);
+	}
+
+	/// <summary>
+	/// Method to CleanUp the resources of the <see cref="MauiPopup"/>.
+	/// </summary>
+	public void CleanUp()
+	{
+		if (VirtualView is null)
+		{
+			return;
+		}
+
+		VirtualView = null;
+
+		var presentationController = (UIPopoverPresentationController)PresentationController;
+		if (presentationController is not null)
+		{
+			presentationController.Delegate = null;
+		}
+	}
+
+	PageHandler CreateControl(in IPopup virtualView)
+	{
+		var view = (View?)virtualView.Content;
+		var contentPage = new ContentPage { Content = view, BackgroundColor = Colors.Orange };
```

Comment:
We can't reference `ContentPage` in `CommunityToolkit.Maui.Core` because `ContentPage` is in the `Microsoft.Maui.Controls` namespace